daycount_test.go

```diff
@@ -287,3 +290,31 @@ func Test_daysPerYear(t *testing.T) {
 	assert.InDelta(t, threeSixtyFiveDays, daysPerYear(2015), tol)
 	assert.InDelta(t, threeSixtySixDays, daysPerYear(2000), tol)
 }
+
+func Test_YearFraction_inverted(t *testing.T) {
+	t.Parallel()
+
+	const tol = 1e-10
+
+	props := gopter.NewProperties(gopter.DefaultTestParametersWithSeed(1234))
```

Comment:
oh wow! A little revolution in our testing habits. Do we use gopter anywhere ATM? I haven't looked a this package yet. It looks like fuzz testing at play, correct?

Couldn't you test your new feature by simply adding test cases onto `Test_YearFraction`?

Reply:
`gopter` is already used in a few repos such as [Eve](https://github.com/edgelaboratories/eve/blob/432eb5929870a5b1275bcdb95a11fe338c2d0b73/internal/model/floating_rate_curve_representation_test.go#L374), [Arcanist](https://github.com/edgelaboratories/arcanist/blob/3586c26d55bf775f32022113827026e9263bfc4b/internal/proxy/seed_test.go#L12C9-L12C9) , [Voldorak](https://github.com/edgelaboratories/voldorak/blob/1353ae22ba758a8f394571f78da67f0c49f8d274/internal/marketdata/timeseries_test.go#L102) , [Etymologist](https://github.com/edgelaboratories/eve/blob/432eb5929870a5b1275bcdb95a11fe338c2d0b73/internal/volatility/moneyness_test.go#L12)...

The function I changed was already tested correctly and was just missing a good test for inverted `from` and `to`. I took the occasion to provide for a better covering test that checks every available convention in the repo for this property :+1:

Reply:
@RemiMattheyDoret to me this here is property testing.
I don't know what fuzz testing means in Go, but I imagine it is a kind of mutation testing ? Where the code is modified on-the-fly to catch untested cases (mutants). Is that what it's about ?

Reply:
OK, I went and search for `gopter` and `gofuzz`. So Gopter is indeed a property testing framework.
And `gofuzz` is not exactly what I was mentioning, it helps populating test objects with random values trying to find edgecases

Reply:
ah indeed! Good to know, thanks. I'll study [gopter](https://pkg.go.dev/github.com/leanovate/gopter) a bit! In the meantime, I'll just trust your test without really reviewing it!